File: php/src/Amen/Resources/Lookups.php
<?php
declare(strict_types=1);
namespace Amen\Resources;
use Amen\AmenClient;
final class Lookups
{
    public function __construct(private AmenClient $c) {}
    public function countryCodes(): array { return $this->c->request('GET', '/allowed-country-codes/'); }
    public function cities(): array { return $this->c->request('GET', '/cities'); }
    public function categories(): array { return $this->c->request('GET', '/categories/'); }
    public function disputeReasons(): array { return $this->c->request('GET', '/dispute-reasons/'); }
    public function disputeResolutionReasons(): array { return $this->c->request('GET', '/dispute-resolution-reasons/'); }
    public function cancelReasons(?string $partyType = null): array { return $this->c->request('GET', '/cancel-reasons/', ['params' => ['party_type' => $partyType]]); }
}

File: flutter/amen_client/lib/src/resources/lookups.dart
import '../client.dart';

class Lookups {
  final AmenClient _c;
  Lookups(this._c);
  Future<List<dynamic>> countryCodes() async => await _c.request('GET', '/allowed-country-codes/') as List;
  Future<List<dynamic>> cities() async => await _c.request('GET', '/cities') as List;
  Future<List<dynamic>> categories() async => await _c.request('GET', '/categories/') as List;
  Future<List<dynamic>> disputeReasons() async => await _c.request('GET', '/dispute-reasons/') as List;
  Future<List<dynamic>> disputeResolutionReasons() async => await _c.request('GET', '/dispute-resolution-reasons/') as List;
  Future<List<dynamic>> cancelReasons({String? partyType}) async => await _c.request('GET', '/cancel-reasons/', params: {'party_type': partyType}) as List;
}
